### ISSUE TITLE
unknown contract interaction for verified proxy/contracts

### ISSUE THREAD

**Issue by shayanb** [safe-global/safe-wallet-web#3680]
## Bug description
After contract upgrades (same proxy address, upgraded implementation), all outgoing transactions to the proxy contract are shown as `unknown contract interaction` . The contracts are all verified on etherscan. 

## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: Ethereum mainnet

## Steps to reproduce
 1. Go to any gnosis multisig
 2. try to call any functions in the pufETH contract: https://etherscan.io/address/0xd9a442856c234a39a81a089c06451ebaa4306a72
 3. the interactions will show as `unknown contract interaction` 

## Expected result
See the correct contract names

## Screenshots
<img width="743" alt="Screenshot 2024-05-09 at 12 41 20 PM" src="https://github.com/safe-global/safe-wallet-web/assets/309108/b8a58b93-b736-4fdb-a350-0993d59c1c44">

**Comment by bxmmm1** [comment]
Also, the TX builder fetches the ERC1967Proxy ABI 
![image](https://github.com/safe-global/safe-wallet-web/assets/28648109/f01f89ec-178a-45db-94e8-728cc6b9e167)

**Comment by katspaugh** [comment]
Pointed out by another reporter of the same issue:

> Probably related: both in this case and in the linked issue we have that the interaction is with a proxy and needs to use the implementation ABI, not the proxy ABI.

**Comment by bxmmm1** [comment]
@katspaugh Add a feature on the frontend to remember the ABI for a certain contract?

**Comment by katspaugh** [comment]
That would only work on the same computer tho, wouldn't it?
Besides, the Transaction Builder app is a separate app on a separate domain, so they don't share any storage with the wallet app.

**Comment by bxmmm1** [comment]
> That would only work on the same computer tho, wouldn't it? Besides, the Transaction Builder app is a separate app on a separate domain, so they don't share any storage with the wallet app.

That is true, but also enough. I think that a lot of web3 teams do their contract management using the Safe UI. It would be really cool to upgrade the **AddressBook** to include the ABI.
That way I could have 5 EOA accounds without an ABI, and a few contracts that are named  & have the ABI stored in the browser storage.
And make it exportable as a JSON, so that the multisig members can share it on slack/discord/whatever.

**Comment by katspaugh** [comment]
We’ll consider this idea, thank you!
